File: 0Highlighted Projects/software-engineering-lab-project/api/TapaBuracos.Core/Models/IssueNotification.cs
using Newtonsoft.Json;
using TapaBuracos.Core.Enums;
using TapaBuracos.Core.Interfaces.Models;

namespace TapaBuracos.Core.Models;

public record IssueNotification : IIssueNotification
{
    [JsonConstructor]
    public IssueNotification(string id, string issueId, string issueDescription, IssueType issueType,
        string description, IssueStatus issueStatus, string userId, bool readNotification, DateTimeOffset createdDate)
    {
        Id = id;
        IssueId = issueId;
        IssueDescription = issueDescription;
        IssueType = issueType;
        Description = description;
        IssueStatus = issueStatus;
        UserId = userId;
        ReadNotification = readNotification;
        CreatedDate = createdDate;
    }
    
    public string Id { get; }
    public string IssueId { get; }
    public string IssueDescription { get; }
    public IssueType IssueType { get; }
    public string Description { get; }
    public IssueStatus IssueStatus { get; }
    public string UserId { get; }
    public bool ReadNotification { get; }
    public DateTimeOffset CreatedDate { get; }
}

File: 0Highlighted Projects/software-engineering-lab-project/app/lib/Models/issue_notification.dart
import 'issue_model.dart';

class IssueNotification {
  final String id;
  final String issueId;
  final String issueDescription;
  final IssueType issueType;
  final String description;
  final IssueStatus issueStatus;
  final String userId;
  final bool readNotification;
  final DateTime createdDate;

  IssueNotification({
    required this.id,
    required this.issueId,
    required this.issueDescription,
    required this.issueType,
    required this.description,
    required this.issueStatus,
    required this.userId,
    required this.readNotification,
    required this.createdDate
  });

  factory IssueNotification.fromMap(Map<String, dynamic> map) {
    return IssueNotification(
        id: map['id'] as String,
        issueId: map['issueId'] as String,
        issueDescription: map['issueDescription'] as String,
        issueType: IssueType.values[map['issueType'] as int],
        description: map['description'] as String,
        issueStatus: IssueStatus.values[map['issueStatus'] as int],
        userId: map['userId'] as String,
        readNotification: map['readNotification'] as bool,
        createdDate: DateTime.parse(map['createdDate'] as String));
  }
}